File: TypeScript/function.ts
function main(): void {
    // Call the function functionImput with the argument "Hello World"
    functionImput("Hello, World!")
    // Call the function functionReturn with the argument "Hello World"
    console.log(functionReturn("Hello, World!"))
    // Call the function that does some math and than returns the value.
    console.log(doSomeMath(2, 3))
    // Call the function functionReturnBoolean
    console.log(functionReturnBoolean())
    // Call the function functionReturnArray
    console.log(functionReturnArray())
}

main()

// Create a function that takes an argument and prints it.
function functionImput(content: string): void {
    console.log(content)
}

// Create a function that takes an argument and returns it.
function functionReturn(content: string): string {
    return content
}

// Create a function that takes two arguments and does some math and than returns
function doSomeMath(firstNumber: number, secondNumber: number): number {
    return firstNumber * secondNumber + secondNumber ^ firstNumber % secondNumber
}

// Create a function that takes no arguments and returns a boolean.
function functionReturnBoolean(): boolean {
    return true
}

// Create a function that takes no arguments and returns an array.
function functionReturnArray(): Array<string> {
    return ["Hello", "World"]
}

// Create a function that takes no arguments and returns a map.
function functionReturnMap(): Map<string, string> {
    return new Map<string, string>([
        ["key1", "value1"],
        ["key2", "value2"]
    ])
}

File: Swift/function.swift
// Simple function no arguments and no returns
func noArgumentsNoReturns() {
    print("No arguments no returns")
}

// Simple function with arguments and no returns
func oneArgumentsNoReturns(content: String) {
    print("One argument no returns: \(content)")
}

// Simple function with arguments and returns
func oneArgumentsReturns(content: String) -> String {
    return "One argument returns: \(content)"
}

// Simple function with two arguments and no returns
func twoArgumentsNoReturns(content1: String, content2: String) {
    print("Two arguments no returns: \(content1), \(content2)")
}

// Simple function with two arguments and returns
func twoArgumentsReturns(content1: String, content2: String) -> (String, String) {
    return ((content1), (content2))
}

func main() {
    // No arguments no returns
    noArgumentsNoReturns()
    // One argument no returns
    oneArgumentsNoReturns(content: "Hello")
    // One argument returns
    print(oneArgumentsReturns(content: "Hello"))
    // Two arguments no returns
    twoArgumentsNoReturns(content1: "Hello", content2: "World")
    // Two arguments returns
    print(twoArgumentsReturns(content1: "Hello", content2: "World"))
}

main()